app.rb

```diff
@@ -1,133 +1,110 @@
-    end
-  end
-
-  def create_teacher
-    puts 'Create a new teacher'
-    print 'Enter teacher age: '
-    age = gets.chomp.to_i
-    print 'Enter teacher name: '
-    name = gets.chomp
-    print 'Enter teacher specialization: '
-    specialization = gets.chomp
-    teacher = Teacher.new(specialization, age, name)
-    @persons.push(teacher)
-    puts 'Teacher created successfully'
-  end
-
-  def create_book()
-    puts 'Create a new book'
-    print 'Enter title: '
-    title = gets.chomp
-    print 'Enter author: '
-    author = gets
-    book = Book.new(title, author)
-    @books.push(book)
-    puts "Book #{title} created successfully."
-  end
-
-  def create_rental
-    puts 'Select which book you want to rent by entering its number'
-    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
-
-    book_id = gets.chomp.to_i
-
-    puts 'Select a person from the list by its number'
-    @persons.each_with_index do |person, index|
-      puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
-    end
-
-    person_id = gets.chomp.to_i
-
+  def create_rental()
+    puts 'Select a book from the following list by the number:📖'
+    list_all_books
+    book_number = gets.chomp.to_i
+    puts ' '
+    puts 'Select a person from the following list by the number:🧑‍🎓🧑‍🏫'
+    list_all_people
+    person_number = gets.chomp.to_i
     print 'Date: '
-    date = gets.chomp.to_s
-
-    rental = Rental.new(date, @persons[person_id], @books[book_id])
-    @rentals << rental
-
-    puts 'Rental created successfully'
+    date = gets.chomp
+    @rentals.push(Rental.new(date, @people[person_number - 1], @books[book_number - 1]))
+    puts 'Rental created successfully 😊'
   end
 
   def list_all_rentals
-    print 'To see person rentals enter the person ID: '
+    print 'ID of person: '
     id = gets.chomp.to_i
-
-    puts 'Rented Books:'
+    puts 'Rentals Books:'
     @rentals.each do |rental|
       if rental.person.id == id
-        puts "Person: #{rental.person.name} Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}"
-
+        puts "Peson: #{rental.person.name}  Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}"
       else
         puts
-        puts 'No records where found for the given ID'
+        puts 'No Rentals found for the given ID'
       end
     end
   end
+
+  def main_menu
+    puts ' '
+    puts 'Please choose an option by entering a number:
+        1 - List all books
+        2 - List all people
+        3 - Create a person
+        4 - Create a book
+        5 - Create a rental
+        6 - List all rentals for a given person id
+        7 - Exit'
+    puts ' '
+  end
+
+  def menu_selection
+    main_menu
+    selection = gets.chomp.to_i
+    case selection
+    when 1
+      @books.list_all_books
+    when 2
+      list_all_people
+    when 3
+      create_person
```

Comment:
- [ ] Hello @silviatofana Nice job on refactoring but this part should be placed in the `main.rb` file